src/sidebar/components/annotation-header.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+var annotationMetadata = require('../annotation-metadata');
+var memoize = require('../util/memoize');
+var persona = require('../filter/persona');
+
+// @ngInject
+function AnnotationHeaderController(groups, settings, serviceUrl) {
+  var self = this;
+
+  this.user = function () {
+    return self.annotation.user;
+  };
+
+  this.username = function () {
+    return persona.username(self.annotation.user);
+  };
+
+  this.isThirdPartyUser = function () {
+    return persona.isThirdPartyUser(self.annotation.user, settings.authDomain);
```

Comment:
`persona.isThirdPartyUser(self.annotation.user, settings.authDomain)` is duplicated a few times in different components.

I think this is an argument for some kind of annotation domain object that encapsulates everything about an annotation and has these methods like `username` and `isThirdPartyUser` on it.

For example if `self.annotation` here in `AnnotationHeaderController` had `self.annotation.username` and `self.annotation.isThirdPartyUser` properties that encapsulated these calls to `persona` then the `AnnotationHeaderController` would be simpler as it could just read those properties instead of having to make the more complex method calls itself, the duplication between components could be removed, and the dependencies of `AnnotationHeaderController` and other components on `persona` could be removed.

We could end up with a single `Annotation` class that depends on persona, groups, annotationMetadata, etc and then each component just uses an instance of that class. Rather than each component depending on each of persona, groups, annotationMetadata, etc